yt/yt/ytlib/api/native/client_queries.cpp

```diff
@@ -38,6 +38,319 @@ using namespace NYTree;
+    query.Id = partialRecord.Key.QueryId;
+    query.Engine = partialRecord.Engine;
+    query.Query = partialRecord.Query;
+    query.Files = partialRecord.Files.value_or(std::nullopt);
+    query.StartTime = partialRecord.StartTime;
+    query.FinishTime = partialRecord.FinishTime.value_or(std::nullopt);
+    query.Settings = partialRecord.Settings.value_or(TYsonString());
+    query.User = partialRecord.User;
+    query.AccessControlObject = partialRecord.AccessControlObject.value_or(std::nullopt);
+    query.State = partialRecord.State;
+    query.ResultCount = partialRecord.ResultCount.value_or(std::nullopt);
+    query.Progress = partialRecord.Progress.value_or(TYsonString());
+    query.Error = partialRecord.Error.value_or(std::nullopt);
+    query.Annotations = partialRecord.Annotations.value_or(TYsonString());
+
+    IAttributeDictionaryPtr otherAttributes;
+    auto fillIfPresent = [&] (const TString& key, const auto& value) {
+        if (value) {
+            if (!otherAttributes) {
+                otherAttributes = CreateEphemeralAttributes();
+            }
+            otherAttributes->Set(key, *value);
+        }
+    };
+
+    if constexpr (std::is_same_v<std::decay_t<decltype(partialRecord)>, TActiveQueryPartial>) {
+        fillIfPresent("abort_request", partialRecord.AbortRequest.value_or(std::nullopt));
+        fillIfPresent("ping_time", partialRecord.PingTime);
+        fillIfPresent("incarnation", partialRecord.Incarnation);
+        fillIfPresent("assigned_tracker", partialRecord.AssignedTracker);
+    }
+
+    query.OtherAttributes = std::move(otherAttributes);
+
+    return query;
+}
+
+//! Lookup one of query tracker state tables by query id.
+template <class TRecordDescriptor>
+TFuture<typename TRecordDescriptor::TRecordPartial> LookupQueryTrackerRecord(
+    TQueryId queryId,
+    const IClientPtr& client,
+    const TString& tablePath,
+    const TString& tableKind,
+    const std::optional<std::vector<TString>>& lookupKeys,
+    NTransactionClient::TTimestamp timestamp)
+{
+    auto rowBuffer = New<TRowBuffer>();
+    const auto& nameTable = TRecordDescriptor::Get()->GetNameTable();
+
+    TLookupRowsOptions lookupOptions;
+    if (lookupKeys) {
+        std::vector<int> columnIds;
+        for (const auto& key : *lookupKeys) {
+            if (auto columnId = nameTable->FindId(key)) {
+                columnIds.push_back(*columnId);
+            }
+        }
+        lookupOptions.ColumnFilter = TColumnFilter(columnIds);
+    }
+    lookupOptions.Timestamp = timestamp;
+    lookupOptions.EnablePartialResult = true;
+    std::vector keys{
+        TActiveQueryKey{.QueryId = queryId}.ToKey(rowBuffer),
+    };
+    auto asyncLookupResult = client->LookupRows(
+        tablePath,
+        TRecordDescriptor::Get()->GetNameTable(),
+        MakeSharedRange(std::move(keys), rowBuffer),
+        lookupOptions);
+    auto asyncRecord = asyncLookupResult.Apply(BIND([=] (const IUnversionedRowsetPtr& rowset) {
```

Comment:
Original commit uses `TUnversionedLookupRowsResult` from [this refactoring commit](https://github.com/ytsaurus/ytsaurus/commit/01bcd9b04eecd2aa43b311fa9ce7882a73806146). We don't bring that commit to 23.2 and manually use old version `IUnversionedRowsetPtr`